Model/Product/Index/IndexRepository.php

```diff
@@ -178,52 +155,10 @@ public function getByProductIdAndStoreId(int $productId, int $storeId)
         return $results->getFirstItem();
     }
 
-
-    /**
-     * Get list of productIndexes
-     *
-     * @param int $pageSize
-     *
-     * @return IndexSearchResults
-     */
-    public function getFirstPage($pageSize)
-    {
-        /* @var IndexCollection $collection */
-        $collection = $this->indexCollectionFactory->create();
-        $collection->setPageSize($pageSize);
-        $collection->setCurPage(1);
-        $collection->load();
-        /* @var IndexSearchResults $searchResults */
-        $searchResults = $this->indexCollectionFactory->create();
-        $searchResults->setItems($collection->getItems());
-        $searchResults->setTotalCount($collection->getSize());
-
-        // Set collection to be mem dealloc
-        $collection->clear();
-
-        return $searchResults;
-    }
-
-    /**
-     * Returns all entries in product queue
-     *
-     * @return IndexSearchResults
-     */
-    public function getAll()
-    {
-        $collection = $this->indexCollectionFactory->create();
-        $collection->load();
-        $searchResults = $this->indexCollectionFactory->create();
-        $searchResults->setItems($collection->getItems());
-        $searchResults->setTotalCount($collection->getSize());
-
-        return $searchResults;
-    }
-
     /**
      * @param SearchCriteriaInterface $searchCriteria
      *
-     * @return SearchResult
+     * @return IndexSearchResults
```

Comment:
Double-check this. `RepositoryUtil::search` is maked to return `Magento\Framework\Api\Search\SearchResult` to me.
Perhaps: `@return SearchResult|IndexSearchResults`